tower-balance/src/lib.rs

```diff
@@ -88,6 +59,52 @@ pub struct ResponseFuture<F: Future, E>(F, PhantomData<E>);
 
 // ===== impl Balance =====
 
+impl<D> Balance<D, choose::PowerOfTwoChoices>
+where
+    D: Discover,
+    D::Service: Load,
+    <D::Service as Load>::Metric: PartialOrd,
+{
+    /// Chooses services using the [Power of Two Choices][p2c].
+    ///
+    /// This configuration is prefered when a load metric is known.
+    ///
+    /// As described in the [Finagle Guide][finagle]:
+    ///
+    /// > The algorithm randomly picks two services from the set of ready endpoints and
+    /// > selects the least loaded of the two. By repeatedly using this strategy, we can
+    /// > expect a manageable upper bound on the maximum load of any server.
+    /// >
+    /// > The maximum load variance between any two servers is bound by `ln(ln(n))` where
+    /// > `n` is the number of servers in the cluster.
+    ///
+    /// [finagle]: https://twitter.github.io/finagle/guide/Clients.html#power-of-two-choices-p2c-least-loaded
+    /// [p2c]: http://www.eecs.harvard.edu/~michaelm/postscripts/handbook2001.pdf
+    pub fn p2c(discover: D) -> Self {
+        Self::new(discover, choose::PowerOfTwoChoices::default())
+    }
+
+    /// Initializes a P2C load balancer from the provided randomization source.
+    ///
+    /// This may be preferable when an application instantiates many balancers.
+    pub fn p2c_from_rng<R: rand::Rng>(
```

Comment:
Nit: I've been using `with` in this case for naming: `p2c_with_rng`. I think it would be worth to stay consistent. I understand that it is *slightly* different in that `rng` is not being moved to a field, but I think it fits anyway.

Reply:
This naming was inspired by `rand::SmallRng::from_rng` -- but happy to change to be more consistent within tower.